flash/core/data/transforms.py

```diff
@@ -13,11 +13,61 @@
 # limitations under the License.
 from typing import Any, Dict, Mapping, Sequence, Union
 
+import numpy as np
 import torch
-from torch import nn
+from torch import nn, Tensor
 
+from flash.core.data.io.input import DataKeys
 from flash.core.data.utilities.collate import default_collate
 from flash.core.data.utils import convert_to_modules
+from flash.core.utilities.imports import _ALBUMENTATIONS_AVAILABLE, requires
+
+if _ALBUMENTATIONS_AVAILABLE:
+    from albumentations import BasicTransform, Compose
+    from albumentations.pytorch import ToTensorV2
+else:
+    BasicTransform, Compose, ToTensorV2 = object, object, object
+
+
+class AlbumentationsAdapter(nn.Module):
+    # mapping from albumentations to Flash
+    TRANSFORM_INPUT_MAPPING = {"image": DataKeys.INPUT, "mask": DataKeys.TARGET}
+
+    @requires("albumentations")
+    def __init__(
+        self,
+        transform: Union[BasicTransform, Sequence[BasicTransform]],
+        mapping: dict = None,
+        image_key: str = DataKeys.INPUT,
+    ):
+        super().__init__()
+        if not isinstance(transform, (list, tuple)):
+            transform = [transform]
+        self.transform = Compose(list(transform) + [ToTensorV2()])
+        self._img_key = image_key
+        if not mapping:
+            mapping = self.TRANSFORM_INPUT_MAPPING
+        self._mapping_rev = mapping
+        self._mapping = {v: k for k, v in mapping.items()}
+
+    def _image_transform(self, x: Tensor) -> np.ndarray:
+        if x.ndim == 3 and x.shape[0] < 4:
```

Comment:
Is the input always a tensor here? For e.g. image classification we use a PIL Image. Maybe we should take this opportunity to load the segmentation images and masks as PIL and NP array respectively. That way we can get rid of some of the conversion logic in here.

Reply:
Good point, need to check some wider type expectations...